app/src/main/java/com/todoist_android/ui/auth/SignupFragment.kt

```diff
@@ -1,9 +1,111 @@
 package com.todoist_android.ui.auth
 
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.navigation.findNavController
+import com.todoist_android.R
+import com.todoist_android.data.network.APIResource
+import com.todoist_android.databinding.FragmentSignupBinding
+import com.todoist_android.view.validateEmail
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collect
 
 @AndroidEntryPoint
 class SignupFragment : Fragment() {
+    private val viewModel: AuthenticationViewModel by viewModels()
+    private lateinit var binding: FragmentSignupBinding
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentSignupBinding.inflate(layoutInflater,container,false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED){
+                viewModel.signupResponse.collect {
+                    when(it){
+                        is APIResource.Success->{
+                            view.findNavController().navigate(R.id.action_signupFragment_to_loginFragment)
+                            binding.progressbarTwo.visibility = View.GONE
+                            Toast.makeText(requireContext(),it.toString(),Toast.LENGTH_SHORT).show()
```

Comment:
`Toast.makeText(requireContext(),"Signup successful! Please login",Toast.LENGTH_LONG).show()`

Reply:
Let me add this